### ISSUE TITLE
[Question] Why comment on interface method not be parsed?

### ISSUE THREAD

**Issue by zedongh** [swaggo/swag#1063]
For interface and implementation seperation, doc comment on interface not concrete implementation should be a right choice.
But i find swag not parse, how it works?

**Comment by zedongh** [comment]
> For interface and implementation seperation, doc comment on interface not concrete implementation should be a right choice. But i find swag not parse, how it works?
> 
> Here is my code: (comment on concrete implementation works)
> 
> ```go
> type HealthProbeController interface {
> 	// Ping godoc
> 	// @Summary Server health probe
> 	// @Description server running health check
> 	// @Tags probe
> 	// @Produce json
> 	// @Success 200 {object} PingResponse
> 	// @Router /api/v1/ping [get]
> 	Ping(c *gin.Context)
> }
> ```

It's seems refer to parser.go
```
// ParseRouterAPIInfo parses router api info for given astFile.
func (parser *Parser) ParseRouterAPIInfo(fileName string, astFile *ast.File) error {
	for _, astDescription := range astFile.Decls {
		astDeclaration, ok := astDescription.(*ast.FuncDecl)
		if ok && astDeclaration.Doc != nil && astDeclaration.Doc.List != nil {
        ....
}
```

only handle `astFile.Decl` match `*ast.FuncDecl case`

**Comment by ubogdan** [comment]
@zedongh 
Interface declarations cannot be used as response definitions/values. We don't mind being more flexible here but we need to implement this in such a way it returns a value from it.

**Comment by zedongh** [comment]
> values

`interface` method defined input argement and return values  much better. 

Buf for gin web framework handler, the return value always be void, response always be put into `*gin.Context`, so 
`@Success 200 {object} PingResponse` comment is same need for interface way and implementation way.

**Comment by ubogdan** [comment]
@zedongh did you tried?
```go
// Ping godoc
// @Summary Server health probe
// @Description server running health check
// @Tags probe
// @Produce json
// @Success 200 {object} PingResponse
// @Router /api/v1/ping [get]
type HealthProbeController interface {
	Ping(c *gin.Context)
}
```

**Comment by ubogdan** [comment]
Swagger expects the route comment to be placed in the main document, not in the type declaration.

**Comment by zedongh** [comment]
> declaration



> @zedongh did you tried?
> 
> ```go
> // Ping godoc
> // @Summary Server health probe
> // @Description server running health check
> // @Tags probe
> // @Produce json
> // @Success 200 {object} PingResponse
> // @Router /api/v1/ping [get]
> type HealthProbeController interface {
> 	Ping(c *gin.Context)
> }
> ```

Yeah, I tried. General info will be parsed, but router api common only be parsed when it commented on function type.

**Comment by ubogdan** [comment]
> > declaration
> 
> > @zedongh did you tried?
> > ```go
> > // Ping godoc
> > // @Summary Server health probe
> > // @Description server running health check
> > // @Tags probe
> > // @Produce json
> > // @Success 200 {object} PingResponse
> > // @Router /api/v1/ping [get]
> > type HealthProbeController interface {
> > 	Ping(c *gin.Context)
> > }
> > ```
> 
> Yeah, I tried. General info will be parsed, but router api comment only be parsed when it commented on function type.

That's true, see [source code](
https://github.com/swaggo/swag/blob/75e9d0156a952d2f090bcdc0ca88f2e2a4925313/parser.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L706).

**Comment by Luna-CY** [comment]
Are there any new developments on this issue? Has it been explicitly rejected or is there a plan for it?

**Comment by ubogdan** [comment]
No active development on this.
